main.js

```diff
@@ -170,3 +170,62 @@ function validationForm(event) {
 submitBtn.addEventListener('click', validationForm);
 
 // -------Start Section Form Validations-----------------//
+
+// -------Start Local Storage Section ------------------//
+
+// Const
+
+const formName = contactForm.querySelector('#name');
+const formEmail = contactForm.querySelector('#email-input');
+const formMsg = contactForm.querySelector('#user-message');
+
+// create an object to save data from form;
+const formData = {
+  name: '',
+  email: '',
+  message: '',
+};
+// first we check if available localStorage, need to create a function to do so
+
+function displayFromLocalStorage() {
+  document.querySelector('#name').value = formData.name;
+  document.querySelector('#email-input').value = formData.email;
+  document.querySelector('#user-message').value = formData.message;
+}
+
+function getFromLocalStorage() {
+  const currentData = localStorage.getItem('formData');
+  const getFormData = JSON.parse(currentData);
+  formData.name = getFormData.name;
+  formData.email = getFormData.email;
+  formData.message = getFormData.message;
+  displayFromLocalStorage();
+}
+
+function saveInLocalStorage() {
+  localStorage.setItem('formData', JSON.stringify(formData));
+  getFromLocalStorage();
+}
+
+function saveFromInput() {
+  formData.name = formName.value;
+  formData.email = formEmail.value;
+  formData.message = formMsg.value;
+  saveInLocalStorage();
+}
+
+function validateLocalStorage() {
+  if (!localStorage.getItem('formData')) {
+    saveFromInput();
+  } else {
+    getFromLocalStorage();
+  }
+}
+
+validateLocalStorage();
+
+formName.onchange = saveFromInput;
+formEmail.onchange = saveFromInput;
+formMsg.onchange = saveFromInput;
```

Comment:
- [OPTIONAL] I'll suggest changing these events from `onchange` to `onkeydown`, currently the user needs to click out of the input for the event to trigger, but using the `onkeydown` event will save to local storage as the user types in the input field. :smile: :wave:

Reply:
I really do appreciate the time to check my code! Will do that for sure! have a good day!